### ISSUE TITLE
MDI window buttons not working

### ISSUE THREAD

**Issue by pawelsalawa** [pawelsalawa/sqlitestudio#2722]
_(This issue was migrated from the old bug tracker of SQLiteStudio)_

Original ID from old bug tracker: 2835
Originally created at: Mon Apr 13 15:56:46 2015
Originally last updated at: Mon Apr 13 15:56:46 2015

MDI window buttons (minimize, maximize, close) not working (click has no effect), though window context menu and keyboard shortcuts (at least ^W) are working fine. 

First time I''ve noticed that on 3.0.3 linux portable, but build from sources 3.0.3 and 3.0.4 has same issue.\n\n**Operating system:**\nLinux 64-bit\n\n**Version:**\n3.0.4

**Comment by pawelsalawa** [comment]
Originally created at: Mon Apr 13 15:56:46 2015

Just to be more clear &acirc; when maximized, MDI window buttons (which are on the same line as main menu) are working.

**Comment by pawelsalawa** [comment]
Originally created at: Mon Apr 13 15:56:46 2015

I'm confused. So the problem appears only when MDI windows are not maximized, right?

If yes, please also tell me on which Linux exactly is this happening?

**Comment by pawelsalawa** [comment]
Originally created at: Mon Apr 13 15:56:46 2015

I'm confused too. Maybe it is a false alarm, caused by Qt (which seems vulnerable to CFLAGS and whatever), so I sheduled rebuild for all dependencies. If it doesn't help I've to do some debugging

**Comment by pawelsalawa** [comment]
Originally created at: Mon Apr 13 15:56:46 2015

Everything related is rebuilt, but with no luck - buttons still not working. Any hints for debug?
Using: Linux 3.18.11-aufs-Apr16 #1 SMP Thu Apr 16 02:14:48 GenuineIntel GNU/Linux
Qt stuff is 5.4.1

**Comment by pawelsalawa** [comment]
Originally created at: Mon Apr 13 15:56:46 2015

What is your Linux distribution?

**Comment by pawelsalawa** [comment]
Originally created at: Mon Apr 13 15:56:46 2015

Gentoo/Xfce, though I'm not really sure this bug is distro-specific. I suspect the culprit is DE/WM but it is just a guess.

**Comment by pawelsalawa** [comment]
Originally created at: Mon Apr 13 15:56:46 2015

I will install Gentoo on virtual machine and use Xfce to see how it works here.

**Comment by pawelsalawa** [comment]
Originally created at: Mon Apr 13 15:56:46 2015

It is possible that it would be a waste of time - because of source-based distro nature, we're almost unable to get &quot;equal&quot; installations (different versions of installed packages, different flags and even arch...). I think about debug version of module with relevant event handler or maybe you can give me a hint how to debug Qt apps. At first, we should find out whether event isn't fired or there is some exception in handler..

**Comment by pawelsalawa** [comment]
Originally created at: Mon Apr 13 15:56:46 2015

Are you compiling SQLiteStudio by yourself, or are you using binary package from sqlitestudio.pl?

**Comment by pawelsalawa** [comment]
Originally created at: Mon Apr 13 15:56:46 2015

All what I've tried:
3.0.3 linux 64bit portable, 
3.0.3 linux 64bit compiled
3.0.4 linux 64bit compiled
3.0.5 linux 64bit portable (just checked)

got same &quot;non-working&quot; buttons

**Comment by pawelsalawa** [comment]
Originally created at: Mon Apr 13 15:56:46 2015

Please run SQLiteStudio from command line with option --debug-stdout
Then click on the problematic &quot;close&quot; button.

Then see if there is anything disturbing in the console (you can paste it here, so we can both look).

**Comment by pawelsalawa** [comment]
Originally created at: Mon Apr 13 15:56:46 2015

I did it before, nothing relevant found. Nevertheless, here's debug output: https://paste.lugons.org/show/8617/

Suddenly, I found the way to click these &quot;unclickable&quot; buttons: right-click to MDI window icon and immediately (while window menu is active) click on any button. Button is really &quot;pressed&quot; (greyed) and do appropriate action. Though, right-click on window caption (same window menu appeared) do not bring buttons into working state :)
